### ISSUE TITLE
Should BodyInterface::MoveKinematic check for SucceededAndIsInBroadPhase ?

### ISSUE THREAD

**Issue by jankrassnigg** [jrouwe/JoltPhysics#847]
In rare cases I get an assert in BodyManager::ActivateBodies():
JPH_ASSERT(body.IsInBroadPhase());

The thing is, I am calling BodyInterface::MoveKinematic() which takes a BodyID and locks the body itself, but it only calls lock.Succeeded(), not SucceededAndIsInBroadPhase().

If I need to check for this before, that would defeat the purpose of going through this interface, as far as I can see.

On the other hand, I am wondering whether BodyManager::ActivateBodies() should assert this, at all? If the body isn't in the broadphase yet, the activation can just be ignored, no?

**Comment by jrouwe** [comment]
The question is: Why call MoveKinematic on a body that hasn't been added to the world yet? Is that not something that you can easily avoid?

If we were to change that Succeeded to a SucceededAndIsInBroadPhase then the call would fizzle completely, which is probably also not what you wanted.

I kind of like the idea that ActivateBodies asserts when you try to activate a body that hasn't been added yet since it feels like a programming error.

**Comment by jankrassnigg** [comment]
Hm, I guess you are right. In my case it turned out to be a very specific situation. I run the simulation at a fixed framerate (say 60 Hz) but if the engine runs faster, I sometimes skip physics updates. Additionally I can't always add Jolt bodies at the time where I create them, so I queue the "add" for later, when it is save. Usually all this works fine, but now the character controller code DOES get updated every frame, even if physics simulation is skipped and THAT one re-creates a "presense body" when you duck / stand-up. And then it could happen that it tried to move the body in the next frame, but it had not been added yet....

Long story short, I found a light-weight solution to test whether a body is still queued for addition, to skip the MoveKinematic.
